File: app/src/main/java/com/erencol/sermon/view/adapters/ViewHolders/SermonViewHolder.java
package com.erencol.sermon.view.adapters.ViewHolders;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.recyclerview.widget.RecyclerView;

import com.erencol.sermon.model.Sermon;
import com.erencol.sermon.databinding.SermonCellBinding;
import com.erencol.sermon.viewmodelpkg.SermonCellViewModel;

public class SermonViewHolder extends RecyclerView.ViewHolder {
    SermonCellBinding sermonCellBinding;
    public View view;

    public SermonViewHolder(SermonCellBinding sermonCellBinding) {
        super(sermonCellBinding.itemSermon);
        this.sermonCellBinding = sermonCellBinding;
    }

    public void bindSermon(Sermon sermon) {
        bindSermon(sermon, false, 0);
    }

    public void bindSermon(Sermon sermon, boolean isLocked, int position) {
        if (sermonCellBinding.getSermonCellViewModel() == null) {
            sermonCellBinding.setSermonCellViewModel(
                    new SermonCellViewModel(sermon, isLocked, position));
        } else {
            sermonCellBinding.getSermonCellViewModel().setSermon(sermon, isLocked, position);
        }
        if(sermon.isNew)
            sermonCellBinding.newAlert.setVisibility(View.VISIBLE);
        else
            sermonCellBinding.newAlert.setVisibility(View.GONE);

    }

}

File: app/src/main/java/com/erencol/sermon/view/main/SermonViewHolder.kt
package com.erencol.sermon.view.main

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView
import com.erencol.sermon.databinding.SermonCellBinding
import com.erencol.sermon.model.Sermon
import com.erencol.sermon.view.main.SermonCellViewModel

class SermonViewHolder(private val sermonCellBinding: SermonCellBinding) :
    RecyclerView.ViewHolder(sermonCellBinding.itemSermon) {
    var view: View? = null
    fun bindSermon(sermon: Sermon, isLocked: Boolean, onLockedClick: (() -> Unit)?) {
        if (sermonCellBinding.sermonCellViewModel == null) {
            sermonCellBinding.sermonCellViewModel = SermonCellViewModel(sermon)
        } else {
            sermonCellBinding.sermonCellViewModel!!.setSermonData(sermon)
        }
        sermonCellBinding.sermonCellViewModel!!.isLocked = isLocked
        sermonCellBinding.sermonCellViewModel!!.onLockedClick = onLockedClick

        if (sermon.isNew == true) {
            sermonCellBinding.newAlert.visibility = View.VISIBLE
        } else {
            sermonCellBinding.newAlert.visibility = View.GONE
        }
    }
}